clippy_lints/src/inline_format_args.rs

```diff
@@ -0,0 +1,80 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::macros::{is_aliased_format_arg, is_format_macro, root_macro_call_first_node, FormatArgsExpn};
+use clippy_utils::source::{expand_past_previous_comma, snippet};
+
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, Path, QPath};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// ### Why is this bad?
+    ///
+    /// ### Example
+    /// ```rust
+    /// // example code where clippy issues a warning
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// // example code which does not raise clippy warning
+    /// ```
+    #[clippy::version = "1.64.0"]
+    pub INLINE_FORMAT_ARGS,
```

Comment:
The lint name is wrong. It suggests that this **lints on** inline format args rather than that it **suggests** using them.

Maybe `needless_explicit_format_args`? :thinking:

Reply:
ah, naming... :)  Maybe `explicit_format_args` is a bit simpler?  or `uninlined_format_args` / `noninlined_format_args`?

Reply:
I'm not the best at lint naming. Let's decide this right before merging. We can also ask other on Zulip by then.